test/integrationtests/session_test.go

```diff
@@ -0,0 +1,231 @@
+	sessionNonce2 := "sessionNonce2"
+	session2 := sessionmanager.PeerSession{
+		IsAuthenticated: false,
+		SessionNonce:    &sessionNonce2,
+		PeerNonce:       &nonce2,
+		PeerIdentityKey: &identityKey,
+		LastUpdate:      time.Now(),
+	}
+	sm.AddSession(session2)
+
+	// when
+	s1 := sm.GetSession(sessionNonce1)
+	s2 := sm.GetSession(sessionNonce2)
+	byIdentity := sm.GetSession(identityKey)
+
+	// then
+	require.NotNil(t, s1)
+	require.Equal(t, nonce1, *s1.PeerNonce)
+
+	require.NotNil(t, s2)
+	require.Equal(t, nonce2, *s2.PeerNonce)
+
+	require.NotNil(t, byIdentity)
+	require.Equal(t, *session2.SessionNonce, *byIdentity.SessionNonce)
+
+	// when
+	session1.IsAuthenticated = true
+	sm.UpdateSession(session1)
+	byIdentityAfterAuth := sm.GetSession(identityKey)
+
+	// then
+	require.NotNil(t, byIdentityAfterAuth)
+	require.Equal(t, *session1.SessionNonce, *byIdentityAfterAuth.SessionNonce)
+
+	// when
+	session2.IsAuthenticated = true
+	sm.UpdateSession(session2)
+	byIdentityBothAuth := sm.GetSession(identityKey)
+
+	// then
+	require.NotNil(t, byIdentityBothAuth)
+	require.Equal(t, *session2.SessionNonce, *byIdentityBothAuth.SessionNonce)
+}
+
+func TestSessionLookupFailure(t *testing.T) {
+	// given
+	sm := sessionmanager.NewSessionManager()
+	nonExistentSessionNonce := "nonexistent-session-nonce"
+	nonExistentIdentityKey := "nonexistent-identity-key"
+
+	// when
+	result1 := sm.GetSession(nonExistentSessionNonce)
+	result2 := sm.GetSession(nonExistentIdentityKey)
+
+	// then
+	require.Nil(t, result1)
+	require.Nil(t, result2)
+
+	// when
+	fakeSession := sessionmanager.PeerSession{
+		SessionNonce:    &nonExistentSessionNonce,
+		PeerIdentityKey: &nonExistentIdentityKey,
+	}
+	sm.RemoveSession(fakeSession)
+
+	result3 := sm.GetSession(nonExistentSessionNonce)
+
+	// then
+	require.Nil(t, result3)
+}
+
+func TestSessionLastUpdateTracking(t *testing.T) {
+	// given
+	sm := sessionmanager.NewSessionManager()
+	identityKey := "0353854510f675922eb4d1ed3fd044c54d161c85852be5bf8074a8a8b1f2ee5273"
+	nonce := "testNonce"
+	sessionNonce := "sessionNonce"
+
+	initialTimestamp := time.Now().Add(-1 * time.Hour)
+	session := sessionmanager.PeerSession{
+		IsAuthenticated: false,
+		SessionNonce:    &sessionNonce,
+		PeerNonce:       &nonce,
+		PeerIdentityKey: &identityKey,
+		LastUpdate:      initialTimestamp,
+	}
+
+	sm.AddSession(session)
+
+	// when
+	retrievedSession := sm.GetSession(sessionNonce)
+
+	// then
+	require.NotNil(t, retrievedSession)
+	require.WithinDuration(t, initialTimestamp, retrievedSession.LastUpdate, time.Millisecond)
+
+	// when
+	time.Sleep(10 * time.Millisecond)
+	updatedTimestamp := time.Now()
+	session.LastUpdate = updatedTimestamp
+	sm.UpdateSession(session)
+	updatedSession := sm.GetSession(sessionNonce)
+
+	// then
+	require.NotNil(t, updatedSession)
+	require.WithinDuration(t, updatedTimestamp, updatedSession.LastUpdate, time.Millisecond)
+	require.True(t, updatedSession.LastUpdate.After(initialTimestamp))
```

Comment:
I don't understand, what are we actually testing here?
That if you set LastUpdate and update the session then LastUpdate will be updated?